Source/towners.cpp

```diff
@@ -354,14 +354,15 @@ void TalkToBarOwner(PlayerStruct &player, TownerStruct &barOwner)
 				InitQTextMsg(TEXT_BANNER2);
 				return;
 			}
-			int i;
-			if (quests[Q_LTBANNER]._qvar2 == 1 && player.HasItem(IDI_BANNER, &i)) {
-				quests[Q_LTBANNER]._qactive = QUEST_DONE;
-				quests[Q_LTBANNER]._qvar1 = 3;
-				player.RemoveInvItem(i);
-				SpawnUnique(UITEM_HARCREST, barOwner.position + DIR_SW);
-				InitQTextMsg(TEXT_BANNER3);
-				return;
+
+			if (quests[Q_LTBANNER]._qvar2 == 1) {
+				if(player.FindAndRemoveInvItem(IDI_BANNER)){

```

Comment:
Why not combine the `if`s with an `&&`? This is creating more nesting for no reason.

I see you did this in quite a few places in this change, so the diff would be drastically reduced if you just combined them.

Reply:
Because I want to emphasize this has to be the last check, otherwise you might end up with removing the item and not meeting all conditions

Reply:
Well... ands already are short circuiting so you have to be aware of the order anyways.

I think this is just a bad change: enforcing order should not cause `if`s to be broken up like this IMHO.